contributors/design-proposals/security-threat-model.md

```diff
@@ -0,0 +1,89 @@
+4. Scheduler controllers assign pods to nodes based on workload definitions and must know details of both to effectively schedule.
+5. Workload controllers create pods that have certain lifecycle rules.
+6. Network providers use information about the topology of the cluster (physical and logical) to give pods IP address and control their access.
+7. Secret controllers create or update secrets for use by users and applications.
+8. Federations need to run workloads on behalf of users on individual clusters.
+9. Other integrations may need access to read or write definitions on some or all of the namespaces in the cluster.
+
+
+## Security boundaries
+
+Several important security boundaries exist in Kubernetes that reduce the possibility of interference or exploit by consumers.
+
+1. Between the container processes within a pod and the node, and between different pods on a node
+2. At the network interfaces between different pods, from the outside of the cluster to inside (ingress), and from the inside of the cluster to outside (egress)
+3. Between infrastructure components and the API
+4. Between end users and the API
+5. Between the API servers and etcd
+6. Between infrastructure monitoring infrastructure and the infrastructure components
+7. Between different tenants, primarily modelled as different namespaces or nodes
+8. Between the federation control plane and the cluster control plane
+
+
+## Multi-tenancy
+
+Kubernetes should be capable of subdividing the cluster for use by distinct participants with their own security boundaries.
+
+1. Single-tenant: A small number of administrators who are also users deploying applications, and are often infrastructure admins
+2. Collaborative Multi-tenant: Two or more teams acting together on the same cluster without strong boundaries between their roles
+3. Multi-tenant: Many users with limited permissions whose applications may need to interact or be isolated, and administrators to configure and enforce that tenancy
+4. Organizational multi-tenant: Several large groups of users and applications (organizations) where the organization admin is granted a subset of cluster administrative permissions in order to subdivide responsibility.
+
+TODO: link to multi-tenancy doc
+
+
+## Threats
+
+The primary threats to Kubernetes are described here:
+
+1. Escaping container and pod isolation to the node or to other containers on the node
+2. Accessing APIs as a user without proper authorization or authentication
+3. Consuming disproportionate resources on the system to deny other users access
+4. Crashing or wedging the system so that no workloads can be processed
+5. Encouraging controllers to act as confused deputies and perform actions at a higher level of trust
+6. Accessing secret data entrusted to the system without appropriate permission for escalating access to the cluster or other systems
+7. Using access to the cluster API to gain elevated permission on the nodes
```

Comment:
goes both ways: also using elevated node permissions to gain elevated API permissions.